quixstreams/app.py

```diff
@@ -263,85 +253,102 @@ def __init__(
             consumer_extra_config.update(quix_app_config.librdkafka_extra_config)
             producer_extra_config.update(quix_app_config.librdkafka_extra_config)
 
-        self._broker_address = broker_address
-        self._consumer_group = consumer_group
-        self._auto_offset_reset = auto_offset_reset
-        self._commit_interval = commit_interval
-        self._commit_every = commit_every
-        self._producer_extra_config = producer_extra_config
-        self._consumer_extra_config = consumer_extra_config
-        self._processing_guarantee = processing_guarantee
-        self._consumer = RowConsumer(
+        self._config = ApplicationConfig(
             broker_address=broker_address,
             consumer_group=consumer_group,
             auto_offset_reset=auto_offset_reset,
+            commit_interval=commit_interval,
+            commit_every=commit_every,
+            # Add default values to the producer config, but allow them to be overwritten
+            # by the provided producer_extra_config dict
+            producer_extra_config={
+                **_default_producer_extra_config,
+                **producer_extra_config,
+            },
+            consumer_extra_config=consumer_extra_config,
+            processing_guarantee=processing_guarantee,
+            consumer_poll_timeout=consumer_poll_timeout,
+            producer_poll_timeout=producer_poll_timeout,
+            auto_create_topics=auto_create_topics,
+            on_consumer_error=on_consumer_error or default_on_consumer_error,
+            on_processing_error=on_processing_error or default_on_processing_error,
+            on_producer_error=on_producer_error or default_on_producer_error,
```

Comment:
I don't know if callbacks belong to this Config object.
There is no sense in re-using them elsewhere except in the `Application` class.

Let's limit the scope of this class only to those values that should be shared.